examples/bert.ipynb

```diff
@@ -0,0 +1,274 @@
+        "id": "ENzdaB8zogBj"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# The HuggingFace model name to use\n",
+        "model_name = \"philschmid/MiniLM-L6-H384-uncased-sst2\"\n",
+        "\n",
+        "# The sentence to run the model on\n",
+        "sentence = \"The quick brown fox jumps over the lazy dog.\"\n",
+        "\n",
+        "print(\"Parsing sentence tokens.\")\n",
+        "example_input = prepare_sentence_tokens(model_name, sentence)\n",
+        "\n",
+        "print(\"Instantiating model.\")\n",
+        "model = OnlyLogitsHuggingFaceModel(model_name)\n",
+        "\n",
+        "print(\"Tracing model.\")\n",
+        "traced = torch.jit.trace(model, example_input)\n",
+        "\n",
+        "print(\"Compiling with Torch-MLIR\")\n",
+        "linalg_on_tensors_mlir = torch_mlir.compile(\n",
+        "    traced,\n",
+        "    example_input,\n",
+        "    output_type=torch_mlir.OutputType.LINALG_ON_TENSORS)\n",
+        "\n",
+        "print(\"Compiling with IREE\")\n",
+        "# Backend options:\n",
+        "#\n",
+        "# llvm-cpu - cpu, native code\n",
+        "# vmvx - cpu, interpreted\n",
+        "# vulkan - GPU for general GPU devices\n",
+        "# cuda - GPU for NVIDIA devices\n",
+        "iree_backend = \"llvm-cpu\"\n",
+        "iree_vmfb = iree_torch.compile_to_vmfb(linalg_on_tensors_mlir, iree_backend)\n",
+        "\n",
+        "print(\"Loading in IREE\")\n",
+        "invoker = iree_torch.load_vmfb(iree_vmfb, iree_backend)\n",
+        "\n",
+        "print(\"Running on IREE\")\n",
+        "result = invoker.forward(example_input)\n",
+        "print(\"RESULT:\", result)"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "b6PqMZfXidbq",
+        "outputId": "65404058-bd73-4401-b5cf-cfc1acbe9f67"
+      },
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Parsing sentence tokens.\n",
+            "Instantiating model.\n",
+            "Tracing model.\n",
+            "Compiling with Torch-MLIR\n",
+            "Compiling with IREE\n",
+            "Loading in IREE\n",
+            "Running on IREE\n",
+            "RESULT: tensor([[ 1.8574, -1.8036]])\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "We are now running our model on IREE.  The compiled version of this model can be saved, deployed, and executed independently of PyTorch.\n",
+        "\n"
+      ],
+      "metadata": {
+        "id": "x3njgRYXnacc"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Runtime Size Comparison\n",
+        "\n",
+        "One benefit of running a model on IREE is lightweight deployment.  The IREE runtime has a significantly smaller footprint than a full PyTorch install."
```

Comment:
The actual runtime is much smaller than that -- ~100kB. the Python packages contribute a lot of overhead. Not sure how to demonstrate that in this notebook though. Only a native C/C++ app will demonstrate it properly.

Reply:
Do you have any examples of how I could set something like that up?  I know it's a "Python" notebook, but it would be nice to be able to demonstrate the 100kB size here too, even if it means running command line compilation calls in these cells.

Reply:
I think you would need to scribble out a .c file and compile it, then link it to the iree runtime. But I don't know if the iree runtime static library is even included in the Python package.

I always imagined that binary size demo to be a separate example with a dedicated .c/.cpp file and probably CMake file.

Reply:
Sounds good.  I'll think about it separately.